src/Domain/Model/Euro.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Domain\Model;
+
+use InvalidArgumentException;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+final class Euro {
+
+	private static $DECIMAL_COUNT = 2;
+	private static $CENTS_PER_EURO = 100;
+
+	private $cents;
+
+	/**
+	 * @param int $cents
+	 * @throws InvalidArgumentException
+	 */
+	public function __construct( int $cents ) {
+		if ( $cents < 0 ) {
+			throw new InvalidArgumentException( 'Amount needs to be positive' );
```

Comment:
Not sure about this. Our use cases require amounts to be positive, in general a currency value might as well be negative. So isn't this something that should be handled by the validators?